easybuild/easyblocks/m/mxnet.py

```diff
@@ -0,0 +1,220 @@
+
+    def prepare_step(self):
+        """Prepare for building and installing MXNet."""
+        super(EB_MXNet, self).prepare_step()
+        self.py_ext.prepare_python()
+
+    def configure_step(self):
+        """Patch 'config.mk' file to use EB stuff"""
+        for (var, env_var) in [('CC', 'CC'), ('CXX', 'CXX'), ('ADD_CFLAGS', 'CFLAGS'), ('ADD_LDFLAGS', 'LDFLAGS')]:
+            self.cfg.update('buildopts', '%s="%s"' % (var, os.getenv(env_var)))
+
+        toolchain_blas = self.toolchain.definition().get('BLAS', None)[0]
+        if toolchain_blas == 'imkl':
+            blas = "mkl"
+            imkl_version = get_software_version('imkl')
+            if LooseVersion(imkl_version) >= LooseVersion('17'):
+                self.cfg.update('buildopts', 'USE_MKL2017=1')
+            self.cfg.update('buildopts', 'MKLML_ROOT="%s"' % os.getenv("MKLROOT"))
+        elif toolchain_blas in ['ACML', 'ATLAS']:
+            blas = "atlas"
+        elif toolchain_blas == 'OpenBLAS':
+            blas = "openblas"
+        elif toolchain_blas is None:
+            raise EasyBuildError("No BLAS library found in the toolchain")
+
+        self.cfg.update('buildopts', 'USE_BLAS="%s"' % blas)
+
+        if get_software_root('NNPACK'):
+            self.cfg.update('buildopts', 'USE_NNPACK=1')
+
+        super(EB_MXNet, self).configure_step()
+
+    def install_step(self):
+        """Specify list of files to copy"""
+        self.cfg['files_to_copy'] = ['bin', 'include', 'lib',
+                                     (['dmlc-core/include/dmlc', 'nnvm/include/nnvm'], 'include')]
+        super(EB_MXNet, self).install_step()
+
+    def extensions_step(self):
+        """Build & Install both Python and R extension"""
+        # we start with the python bindings
+        self.py_ext.src = os.path.join(self.mxnet_src_dir, "python")
+        change_dir(self.py_ext.src)
+
+        self.py_ext.prerun()
+        self.py_ext.run(unpack_src=False)
+        self.py_ext.postrun()
+
+        # next up, the R bindings
+        self.r_ext.src = os.path.join(self.mxnet_src_dir, "R-package")
+        change_dir(self.r_ext.src)
+        mkdir("inst")
+        symlink(os.path.join(self.installdir, "lib"), os.path.join("inst", "libs"))
+        symlink(os.path.join(self.installdir, "include"), os.path.join("inst", "include"))
+
+        # MXNet doesn't provide a list of its R dependencies by default
+        write_file("NAMESPACE", r_namespace)
+        change_dir(self.mxnet_src_dir)
+        self.r_ext.prerun()
+        # MXNet is just weird. To install the R extension, we have to:
+        # - First install the extension like it is
+        # - Let R export the extension again. By doing this, all the dependencies get
+        #   correctly filled and some mappings are done
+        # - Reinstal the exported version
+        self.r_ext.run()
+        run_cmd("R_LIBS=%s Rscript -e \"require(mxnet); mxnet:::mxnet.export(\\\"R-package\\\")\"" % self.installdir)
```

Comment:
If you really insist but it was a pain to get working and I'm sure this one works.